utils_cv/detection/dataset.py

```diff
@@ -96,38 +150,64 @@ def __init__(
             self.train_ds,
             batch_size=self.batch_size,
             shuffle=True,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
         self.test_dl = DataLoader(
             self.test_ds,
             batch_size=self.batch_size,
             shuffle=False,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
-    def _get_labels(self) -> List[str]:
+    def _read_annos(self) -> List[str]:
```

Comment:
can we refactor to
```python

def _read_annos(self) -> List[str]:
        anno_filenames = sorted(os.listdir(self.root / self.anno_dir))
        self.im_paths = []

        # Parse all annotations
        self.anno_paths = []
        self.anno_bboxes = []
        for anno_filename in anno_filenames:
            anno_path = self.root / self.anno_dir / str(anno_filename)
            anno_bboxes, im_path = parse_pascal_voc_anno(anno_path)

            # TODO For now, ignore all images without a single bounding box in it, otherwise throws error during training.
            if len(anno_bboxes) == 0:
                continue

            # get im_path from anno or from im_dir
            if self.im_dir is None:
                self.im_paths.append(im_path)
            else:
                self.im_paths.append(os.path.splitext(anno_path)[0] + ".jpg"

            self.anno_paths.append(anno_path)
            self.anno_bboxes.append(anno_bboxes)

```

Reply:
That's what I had first, but then changed to the current code so that I am not assuming an image ends with ".jpg" and rather I assume the annotation files end with ".xml"